core/src/refx/subs.cljc

```diff
@@ -76,6 +76,38 @@
   (doseq [[_ sub] @sub-cache]
     (dispose! sub)))
 
+(defonce ^:private listeners-state
+  (atom {:counter 0 :pending (sorted-map)}))
+
+(defn- invoke-listener
+  "This function is responsible for ensuring that signal listeners
+  (from DynamicSubs) are called before triggering regular listeners
+  (eg: added via use-sub hook). Listeners are triggered in the order they
+  were registered. This function ensures that one db update will only trigger
+  a single render."
+  [listener-key listener-fn]
+  (swap! listeners-state (fn [state]
+                           (cond-> (update state :counter inc)
+                             (not (signal? listener-key))
+                             (update :pending assoc listener-key listener-fn))))
```

Comment:
This won't properly sort `use-sub-1`, `use-sub-2`, `use-sub-11`. Will get this addressed.